src/NuGet.Clients/NuGet.VisualStudio.OnlineEnvironment.Client/SolutionExplorer/NuGetWorkspaceCommandHandler.cs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Threading;
+using Microsoft.VisualStudio.Workspace.VSIntegration.UI;
+
+namespace NuGet.VisualStudio.OnlineEnvironment.Client
+{
+    /// <summary>
+    /// Extends the Solution Explorer in online environment scenarios.
+    /// </summary>
+    internal class NuGetWorkspaceCommandHandler : IWorkspaceCommandHandler
+    {
+        private readonly JoinableTaskContext _taskContext;
+        private readonly IAsyncServiceProvider _serviceProvider;
+
+        private static bool IsNuGetFunctionalityAvailable = false;
+
+        public NuGetWorkspaceCommandHandler(JoinableTaskContext taskContext, IAsyncServiceProvider asyncServiceProvider)
+        {
+            _taskContext = taskContext;
+            _serviceProvider = asyncServiceProvider;
+        }
+
+        /// <summary>
+        /// The command handlers priority. If there are multiple handlers for a given node
+        /// then they are called in order of decreasing priority.
+        /// </summary>
+        public int Priority => 2000;
```

Comment:
Why 2000? Will 1000 behave differently?

Reply:
Yes, it will get prompted earlier. 

Basically the way this works is that all the handlers are called in priority order until one handles it. 
I was just recommended to use priority 2000 because we don't have that many commands that we are handling, so it's better if we are further down the list.

Reply:
Understand. Probably out of the picture now. But later would be any way to add telemetry to measure how fast they call into us?

Reply:
Given that this is on the UI thread, we have ongoing metrics to handle this scenario.

Reply:
It's a VS level infra, not NuGet specific :)